static/app/components/modals/sudoModal.tsx

```diff
@@ -42,36 +42,63 @@ type State = {
   busy: boolean;
   error: boolean;
   errorType: string;
+  showAccessForms: boolean;
   superuserAccessCategory: string;
   superuserReason: string;
 };
 
 class SudoModal extends Component<Props, State> {
   state: State = {
+    authenticators: [],
+    busy: false,
     error: false,
     errorType: '',
-    busy: false,
+    showAccessForms: true,
     superuserAccessCategory: '',
     superuserReason: '',
-    authenticators: [],
   };
 
   componentDidMount() {
     this.getAuthenticators();
   }
 
-  handleSubmit = async () => {
-    const {api, isSuperuser} = this.props;
-    const data = {
-      isSuperuserModal: isSuperuser,
+  handleSubmitCOPS = () => {
+    this.setState({
       superuserAccessCategory: 'cops_csm',
       superuserReason: 'COPS and CSM use',
-    };
-    try {
-      await api.requestPromise('/auth/', {method: 'PUT', data});
-      this.handleSuccess();
-    } catch (err) {
-      this.handleError(err);
+    });
+  };
+
+  handleSubmit = async data => {
+    const {api, isSuperuser} = this.props;
+    const superuserAccessCategory =
+      this.state.superuserAccessCategory === null
```

Comment:
can we just check falsey-ness?
```
const {superuserAccessCategory, superuserReason} = this.state;
const suAccessCategory = !superuserAccessCategory ? data.superuserAccessCategory : superuserReason;
```
?

Reply:
actually naw.
Lets just say `const suAccessCategory = this.state.superuserAccessCategory || data.superuserAccessCategory`